packages/worker/src/transcribe.ts

```diff
@@ -0,0 +1,190 @@
+import { spawn } from 'child_process';
+import path from 'path';
+import * as fs from 'fs';
+
+interface ProcessResult {
+	code?: number;
+	stdout: string;
+	stderr: string;
+}
+
+interface FfmpegResult {
+	wavPath: string;
+	duration?: number;
+}
+
+const CONTAINER_FOLDER = '/input';
+
+const runSpawnCommand = (
+	cmd: string,
+	args: ReadonlyArray<string>,
+): Promise<ProcessResult> => {
+	return new Promise((resolve, reject) => {
+		const cp = spawn(cmd, args);
+		const stdout: string[] = [];
+		const stderr: string[] = [];
+		cp.stdout.on('data', (data) => {
+			console.log(data.toString());
+			stdout.push(data.toString());
+		});
+
+		cp.stderr.on('data', (data) => {
+			console.log(data.toString());
+			stderr.push(data.toString());
+		});
+
+		cp.on('error', (e) => {
+			stderr.push(e.toString());
+		});
+
+		cp.on('close', (code) => {
+			const result = {
+				stdout: stdout.join(''),
+				stderr: stderr.join(''),
+				code: code || undefined,
+			};
+			if (code === 0) {
+				resolve(result);
+			} else {
+				console.error(
+					`failed with code ${result.code} due to: ${result.stderr}`,
+				);
+				reject(result);
+			}
+		});
+	});
+};
+
+export const createContainer = async (tempDir: string): Promise<string> => {
+	const existingContainer = await runSpawnCommand('docker', [
+		'ps',
+		'--filter',
+		'name=whisper',
+		'--format',
+		'{{.ID}}',
+	]);
+
+	if (existingContainer.stdout) {
+		return existingContainer.stdout.trim();
+	}
+
+	const newContainer = await runSpawnCommand('docker', [
+		'run',
+		'-t',
+		'-d',
+		'--name',
+		'whisper',
+		'-v',
+		`${tempDir}:${CONTAINER_FOLDER}`,
+		'ghcr.io/guardian/transcription-service',
+	]);
+	return newContainer.stdout.trim();
+};
+
+export const convertToWav = async (
+	containerId: string,
+	file: string,
+): Promise<FfmpegResult> => {
+	const fileName = path.basename(file);
+	const filePath = `${CONTAINER_FOLDER}/${fileName}`;
+	const wavPath = `${CONTAINER_FOLDER}/output.wav`;
```

Comment:
maybe we should do ${fileName}-converted.wav rather than output.wav in case the same machine does more than one file so we can tell them apart